packages/api/src/entity/library_item.ts

```diff
@@ -204,4 +204,7 @@ export class LibraryItem {
 
   @Column('text')
   highlightAnnotations?: string[]
+
+  @Column('timestamptz')
+  sharedAt?: Date
```

Comment:
Thinking out loud, but also wonder if this could be `is_discover_item` not as flexible for future use, but maybe more explicit.

Reply:
I think this and above are similar question: _Should we have a join table for things like `discover_item` and `item_embedding`_.

There are several considerations:
1. search performance for inner joins:
Probably not so bad for one-to-one relations like `discover_item` and `item_embedding`. I think we can benchmark the query.

2. frequently update column: I agree with your thought and it has been a big concern for us. Worth to move these columns to a separate table.

3. optimization:
Instead of running cosine similarity query on the fly, we could pre-generate the topics or some other features and store them in another table.

4. naming: 
I think if we want to be more explicit, we should better go with a separate table `discover_item` which links to the `libraryItem` table for normalization and reducing the size of the table.

Having considered all these concerns, I think we should
1. Test and benchmark the search query against join tables (probably only one-to-one relations) in cloned DB
2. If (1) result is not bad, we could create two additional separate tables: `discover_item` and `item_embedding`
3. Create another job to generate topic when the `embedding` is updated and stored in the table

Reply:
Hi @sywhb I also mentioned this on Slack, but a lot of what you've talked about in here is already done as part of my Discover update. 

The optimisation part is already done as part of the import in discover, see: 
https://github.com/omnivore-app/omnivore/blob/main/packages/db/migrations/0168.do.add_discover_feeds.sql

The discover feeds contain already the topics and embeddings. There's also a separate table for discover items, which have their embeddings stored, whether a user has saved a story - which is used to calculate the popularity score in discover too. 

The calculations of the embeddings is done via https://github.com/omnivore-app/omnivore/blob/main/packages/discover/src/index.ts - This deliberately slows down the embedding process so that you never exceeds the RPS that the API supports.

Reply:
Thank you @Podginator ! Taking a look now